### ISSUE TITLE
403 when trying to access an empty documentation even as a Django superuser.

### ISSUE THREAD

**Issue by ayharano** [marcgibbons/django-rest-swagger#589]
Hello there.

I am not sure if it is a real issue, but given that I wasn't able to find a similar one as opened, I will give it a try.

I am trying to set an empty project using cookiecutter-django and DRF 3.5.3 and django-rest-swagger 2.1.0.

I configured the setting with the same options as django-rest-swagger Docker-ready example:

```python
SWAGGER_SETTINGS = {
    'LOGIN_URL': 'rest_framework:login',
    'LOGOUT_URL': 'rest_framework:logout',
    'USE_SESSION_AUTH': True,
    'DOC_EXPANSION': 'list',
    'APIS_SORTER': 'alpha'
}
```

My current DRF settings are:

```python
REST_FRAMEWORK = {
    'PAGE_SIZE': 100,
    'DEFAULT_VERSIONING_CLASS': 'rest_framework.versioning.URLPathVersioning',
    'DEFAULT_PERMISSION_CLASSES': (
        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly',
    )
}
```

My relevant url configuration is:

```python
from django.conf.urls import include, url

from rest_framework.routers import DefaultRouter
from rest_framework_swagger.views import get_swagger_view


title = 'Foobar API'

v1router = DefaultRouter()

swagger_schema_view = get_swagger_view(title=title, url='/docs')

urlpatterns = [
...
    url(r'^api-auth/', include('rest_framework.urls', namespace='rest_framework')),
    url(r'^docs/', swagger_schema_view),
    url(r'^(?P<version>v1)/', include(v1router.urls, namespace='v1')),
...
]
```

Namespace `v1` is empty at this point. I set up a superuser and even after logging in using that superuser, I get a 403 error rendered into django-rest-swagger template:

`403 : {"detail":"You do not have permission to perform this action."} http://IP:8000/docs/?format=openapi`

I still haven't tried to set any endpoint into `v1` namespace. I tried to find out why it wouldn't allow a superuser to access the generated documentation, but I wasn't able to find a clue about it.

Is it an expected behavior of django-rest-swagger?

Beforehand, I would like to say that I'm sorry if this wasn't the proper place to report such behavior. If that is the case, please notify where should I ask for a guidance about this issue, please.

**Comment by mcranston18** [comment]
@ayharano what happens if you add `SessionAuthentication` to your `REST_FRAMEWORK` config?

```
    'DEFAULT_AUTHENTICATION_CLASSES': (
       'rest_framework.authentication.SessionAuthentication',
    ),
```

**Comment by ayharano** [comment]
@mcranston18 I added the setting as suggested, but the issue still persists.

**Comment by marcgibbons** [comment]
@ayharano The issue here is that the schema view is not publicly available by default. I will update the default get_swagger_view() shortcut in a following release to default the schema to AllowAny, this way the spec is always accessible (though permissions will be enforced for the contents of the spec).

In the meantime, I suggest using a custom SwaggerView and setting permission_classes to `AllowAny`.

Example: 

```python
class SchemaView(views.APIView):
    exclude_from_schema = True
    renderer_classes = [
        OpenAPIRenderer,
        SwaggerUIRenderer,
    ]
    permission_classes = [permissions.AllowAny]

    def get(self, request, *args, **kwargs):
        generator = schemas.SchemaGenerator(title='My API')
        schema = generator.get_schema(request)

        return response.Response(schema)
```

**Comment by marcgibbons** [comment]
Resolved in 2.1.1.
